nucleus/common/common-util/src/main/java/com/sun/enterprise/loader/ASURLClassLoader.java

```diff
@@ -118,7 +118,7 @@ public class ASURLClassLoader extends GlassfishUrlClassLoader
     private volatile String doneSnapshot;
 
     /** streams opened by this loader */
-    private final Vector<SentinelInputStream> streams = new Vector<>();
+    private final List<SentinelInputStream> streams = new ArrayList<>(16);
```

Comment:
Potentially unsynchronized access in `SentinelInputStream` constructor or I wrong?

Reply:
@avpinchuk Yep you're right, should be OK now.